### PR TITLE
groups: sample based on azure sdk team updates

### PR DESCRIPTION
**Pull request by idg-sam**
## Purpose

* ...

## Does this introduce a breaking change?

```
[ ] Yes
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

```
```

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

File: 3-Authorization-II/groups/README.md [hunk 1]
```diff
@@ -0,0 +1,468 @@
...
+   - Type a key description (for instance `app secret`),
+   - Select one of the available key durations (**6 months**, **12 months** or **Custom**) as per your security posture.
+   - The generated key value will be displayed when you select the **Add** button. Copy and save the generated value for use in later steps.
+   - You'll need this key later in your code's configuration files. This key value will not be displayed again, and is not retrievable by any other means, so make sure to note it from the Azure portal before navigating to any other screen or blade.
+1. In the app's registration screen, select the **API permissions** blade in the left to open the page where we add access to the APIs that your application needs.
+   - Select the **Add a permission** button and then,
+   - Ensure that the **Microsoft APIs** tab is selected.
+   - In the *Commonly used Microsoft APIs* section, select **Microsoft Graph**
+   - In the **Delegated permissions** section, select the **GroupMember.Read.All** in the list. Use the search box if necessary.
+   - Select the **Add permissions** button at the bottom.
```
Comment (derisen) [inline 3-Authorization-II/groups/README.md:161]: Missing step: need to grant admin consent to this permission
Might also be a good idea to briefly mention this is needed to handle overage scenarios

File: 3-Authorization-II/groups/AppCreationScripts/Steps.md [hunk 2]
```diff
@@ -0,0 +1,318 @@
+---
+page_type: sample
+languages:
+  - csharp
+  - javascript
+  - typescript
+  - python
+  - java
```
Comment (kalyankrishna1) [inline 3-Authorization-II/groups/AppCreationScripts/Steps.md:8]: >java [](http://example.com/codeflow?start=4&length=4)

delete this file

File: 3-Authorization-II/groups/README.md [hunk 3]
```diff
@@ -0,0 +1,468 @@
...
+
+## Contents
+
+| File/folder       | Description                                |
+|-------------------|--------------------------------------------|
+| `AppCreationScripts/`                                                               | Scripts to automatically configure Azure AD app registrations.                              |
+| `pom.xml`                                                                           | Application dependencies.                                                                   |
+| `src/main/resources/templates/`                                                     | Thymeleaf Templates for UI.                                                                 |
+| `src/main/resources/application.yml`                                                | Application and Azure AD Boot Starter Library Configuration.                                |
+| `src/main/java/com/microsoft/azuresamples/msal4j/msidentityspringbootwebapp/`       | This directory contains the main application entry point, controller, and config classes.   |
```
Comment (kalyankrishna1) [inline 3-Authorization-II/groups/README.md:65]: >msidentityspringbootwebapp [](http://example.com/codeflow?start=51&length=26)

nit:line breaks

File: 3-Authorization-II/groups/README.md [hunk 4]
```diff
@@ -0,0 +1,468 @@
...
+| `LICENSE`                                                                           | The license for the sample.                                                                 |
+
+## Prerequisites
+
+- [JDK Version 15](https://jdk.java.net/15/). This sample has been developed on a system with Java 15 but may be compatible with other versions.
+- [Maven 3](https://maven.apache.org/download.cgi)
+- [Java Extension Pack for Visual Studio Code](https://marketplace.visualstudio.com/items?itemName=vscjava.vscode-java-pack) is recommended for running this sample in VSCode.
+- An **Azure AD** tenant. For more information see: [How to get an Azure AD tenant](https://docs.microsoft.com/azure/active-directory/develop/quickstart-create-new-tenant)
+- A user account in your **Azure AD** tenant. This sample will not work with a **personal Microsoft account**. Therefore, if you signed in to the [Azure portal](https://portal.azure.com) with a personal account and have never created a user account in your directory before, you need to do that now.
+- Two security groups, **AdminGroup** and **UserGroup**, containing users you want to test with.
```
Comment (kalyankrishna1) [inline 3-Authorization-II/groups/README.md:81]: >Two security groups [](http://example.com/codeflow?start=2&length=19)

Two security groups "named". If you want to use an existing group then see the section below [write section]

File: 3-Authorization-II/groups/README.md [hunk 5]
```diff
@@ -0,0 +1,468 @@
...
+
+### Create Security Groups
+
+1. Navigate to the [Azure portal](https://portal.azure.com) and select the **Azure AD** service.
+1. Select **Groups** blade on the left.
+1. In the **Groups** blade, select **New Group**.
+    - For **Group Type**, select **Security**
+    - For **Group Name**, enter **AdminGroup**
+    - For **Group Description**, enter **Admin Security Group**
+    - Add **Group Owners** and **Group Members** as you see fit.
```
Comment (kalyankrishna1) [inline 3-Authorization-II/groups/README.md:171]: >see fit [](http://example.com/codeflow?start=56&length=7)

you want to use for this sample

File: 3-Authorization-II/groups/AppCreationScripts/sample.json [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+            "SecurityGroups": [
+                {
+                    "Name": "AdminGroup",
+                    "Description": "Admin Security Group"
+                },
+                {
+                    "Name": "UserGroup",
+                    "Description": "User Security Group"
+                }
+            ]
```
Comment (kalyankrishna1) [inline 3-Authorization-II/groups/AppCreationScripts/sample.json:37]: >] [](http://example.com/codeflow?start=12&length=1)

add manual step. :"Navigate to the portal and provde admin consent to grp..read.all"

File: 3-Authorization-II/groups/src/main/resources/templates/content/403.html [hunk 7]
```diff
@@ -0,0 +1,17 @@
+<div class="card">
+    <h5 class="card-header bg-primary">
+        403: Forbidden
+    </h5>
+    <div class="card-body">
+        <p class="card-text">
+            <strong>Sorry!</strong> you are not a member of the app role(s) that are allowed to visit this page.
```
Comment (kalyankrishna1) [inline 3-Authorization-II/groups/src/main/resources/templates/content/403.html:7]: >app role(s) [](http://example.com/codeflow?start=64&length=11)

groups

**Pull request merged**